packages/core/upload/server/controllers/admin-upload.js

```diff
@@ -0,0 +1,88 @@
+'use strict';
+
+const _ = require('lodash');
+const { ApplicationError } = require('@strapi/utils').errors;
+const { getService } = require('../utils');
+const { ACTIONS } = require('../constants');
+const validateUploadBody = require('./validation/admin/upload');
+const { findEntityAndCheckPermissions } = require('./utils/find-entity-and-check-permissions');
+
+const fileModel = 'plugin::upload.file';
+
+module.exports = {
+  async updateFileInfo(ctx) {
+    const {
+      state: { userAbility, user },
+      query: { id },
+      request: { body },
+    } = ctx;
+
+    const uploadService = getService('upload');
+    const { pm } = await findEntityAndCheckPermissions(userAbility, ACTIONS.update, fileModel, id);
+
+    const data = await validateUploadBody(body);
```

Comment:
21 to 23 :)

Reply:
Ok :) Since 21 and 23 can both throw an error, I would avoid doing them in parallel. Otherwise we can have the case were the same requests receive different response depending on which one throws first

Reply:
I see. Makes sense!